libgnucash/engine/gnc-commodity.cpp

```diff
@@ -470,30 +434,14 @@ gnc_quote_source_get_index (const gnc_quote_source *source)
         return 0;
     }
 
-    switch (source->get_type())
-    {
-    case SOURCE_CURRENCY:
-        return 0;
-    case SOURCE_SINGLE:
-        for (auto i = 0; i < num_single_quote_sources; ++i)
-            if (&single_quote_sources[i] == source)
-                return i;
-        break;
-    case SOURCE_MULTI:
-        for (auto i = 0; i < num_multiple_quote_sources; ++i)
-            if (&multiple_quote_sources[i] == source)
-                return i;
-        break;
-    case SOURCE_UNKNOWN:
-    {
-        for (size_t i = 0; i < new_quote_sources.size(); ++i)
-            if (&new_quote_sources[i] == source)
-                return i;
-        break;
-    }
-    default:
-        break;
-    }
+    auto& sources = get_quote_source_from_type (source->get_type());
+    auto is_source = [&source](const auto& source_iter)
```

Comment:
Calling the parameter to this lambda function `source_iter` is slightly misleading. It's a genuine `gnc_quote_source`, not an iter to one. It set me on the wrong foot for a while...